logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/keypackage/RefillKeyPackageUseCaseTest.kt

```diff
@@ -0,0 +1,86 @@
+package com.wire.kalium.logic.feature.keypackage
+
+import com.wire.kalium.logic.NetworkFailure
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.keypackage.KeyPackageRepository
+import com.wire.kalium.logic.framework.TestClient
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.network.api.keypackage.KeyPackageCountDTO
+import io.mockative.Mock
+import io.mockative.anything
+import io.mockative.classOf
+import io.mockative.eq
+import io.mockative.given
+import io.mockative.mock
+import io.mockative.once
+import io.mockative.verify
+import kotlinx.coroutines.test.runTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertIs
+
+class RefillKeyPackageUseCaseTest {
+
+    @Mock
+    private val keyPackageRepository = mock(classOf<KeyPackageRepository>())
+
+    @Mock
+    private val clientRepository: ClientRepository = mock(classOf())
+
+    private lateinit var refillKeyPackageUseCase: RefillKeyPackagesUseCase
+
+    @BeforeTest
+    fun setup() {
+        refillKeyPackageUseCase = RefillKeyPackagesUseCaseImpl(
+            keyPackageRepository, clientRepository
+        )
+    }
+
+    @Test
+    fun givenKeyPackageCountIs50PercentBelowLimit_ThenRequestToRefillKeyPackageIsPerformed() = runTest {
+        val keyPackageCount = (KEY_PACKAGE_LIMIT * 0.5 - 1).toInt()
+
+        given(clientRepository).suspendFunction(clientRepository::currentClientId).whenInvoked().then { Either.Right(TestClient.CLIENT_ID) }
```

Comment:
Could you please check other tests and the arrangement  practice in cleaning the test classes? Please take a look at AddMemberToConversationUseCaseTest for example.

Reply:
Done